docs/concepts/alternatives.qmd

```diff
@@ -0,0 +1,103 @@
+# Alternatives
+
+Compare/contrast Ibis with other popular dataframe libraries.
+
+| Library | API              | Eager/Lazy | Query Optimizer                      | Collection Types | Engine      | Multi-core | Distributed | Out-of-core | Arrow Backed    |
+| ------- | ---------------- | ---------- | ------------------------------------ | ---------------- | ----------- | ---------- | ----------- | ----------- | --------------- |
+| Ibis    | expression-based | Lazy       | ✅, plus some backends do additional | ✅               | 🟡          | 🟡         | 🟡          | 🟡          | 🟡              |
+| Pandas  | pandas           | Eager      | ❌                                   | Python object    | numpy/arrow | ❌         | ❌          | ❌          | optional >= 2.0 |
+| Dask DF | pandas           | Eager      | ❌                                   | Python object    | numpy/arrow | ✅         | ✅          | ✅          | ❌              |
+| Modin   | pandas           | Eager      | ❌                                   | Python object    | numpy/arrow | ✅         | ✅          | ✅          | ❌              |
+| Vaex    | pandas-close     | Lazy       | ❌                                   | some             | custom      | ✅         | ❌          | ✅          | ❌              |
+| Polars  | expression-based | Lazy       | ✅                                   | ✅               | custom      | ✅         | ❌          | ✅          | ✅              |
+| Daft    | expression-based | Lazy       | ✅                                   | ✅               | custom      | ✅         | ✅          | ✅          | ✅              |
+| Pyspark | expression-based | Lazy       | ✅                                   | ❌               | spark SQL   | ✅         | ✅          | ✅          | ❌              |
+| DuckDB  | SQL              | Eager      | ✅                                   | ✅               | itself      | ✅         | ❌          | ✅          | ✅              |
+
+🟡 = depends on the backend. So if you use the duckdb or Polars backend,
+then you get multi-core, out-of-core, and arrow-backed. Or if you use the
+Spark, BigQuery, Snowflake, etc backend, then you get distributed.
+
+Some general summaries:
+
+Any *eager* library is going to be limited in its ability to optimize queries.
```

Comment:
I think this needs to be reconsidered if DuckDB falls into the eager camp.  They do not appear to be limited in their ability to optimize queries.

Reply:
riiiggght, I'm splitting things up poorly, I meant for all eager imperative APIs